app/src/main/scala/org/alephium/explorer/api/model/ContractParent.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.api.model
+
+import org.alephium.explorer.api.Json._
+import org.alephium.json.Json._
+import org.alephium.protocol.model.Address
+
+@SuppressWarnings(Array("org.wartremover.warts.DefaultArguments"))
+final case class ContractParent(
```

Comment:
Question: would `Option[Address]` be enough? :P

Reply:
Okey, I was thinking about a more Resty way: returning `Address` when the parent exists and 404 when the parent doesn't, and always returning a list of `Address` for subcontracts.
But it is more important to be consistent with other endpoints, if it is the similar approach that it is good :+1:

Reply:
404 is when the contract itself doesn't exist, if you get `None` you know the contract exists but has no parent.